Tensor.lua

```diff
@@ -525,6 +525,25 @@ function Tensor.totable(tensor)
   return result
 end
 
+function Tensor.permute(tensor, ...)
+  local perm = {...}
+  local nDims = tensor:dim()
+  assert(#perm == nDims, 'Invalid permutation')
```

Comment:
How would an under-specified permutation be interpreted then? E.g., 
x = torch.Tensor(1,2,3,4)
y = x:permute(4,3) 
-- not really well-defined and possibly mistaken for transpose(4,3)

For this to work, it would require that the user specified a full permutation on the first n <= N entries (e.g. in the above example, permute(3,1,2) would actually work).
This seemed to me a somewhat questionable usecase and not worth encouraging?